### ISSUE TITLE
become module: ascii exception

### ISSUE THREAD

**Issue by bhcopeland** [ansible/ansible#18444]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
become

##### ANSIBLE VERSION

ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

##### OS / ENVIRONMENT
My OS: Arch 4.8.6
Server: Ubuntu 14.04

##### SUMMARY
5 servers deploy this, it only fails on one. However, this is the only server that runs inside LXC. I have the ACL package installed.

##### STEPS TO REPRODUCE
```

- name: Checkout application code
  git: repo={{django_project_repo}}
       version={{django_project_version}}
       dest={{django_app_root}}
       update=yes
  become: true
  become_user: "{{apache_user}}"
  notify: restart-apache

```

##### EXPECTED RESULTS
Git module should have updated and not thrown an exception. I tried remove {{apache_user}} but made no difference.

##### ACTUAL RESULTS

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 119, in run
    res = self._execute()
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 490, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/normal.py", line 33, in run
    results = merge_hash(results, self._execute_module(tmp=tmp, task_vars=task_vars))
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 617, in _execute_module
    self._fixup_perms2(remote_files, remote_user)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 394, in _fixup_perms2
    ' see https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user'.format(res['rc'], res['stderr']))
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2018' in position 29: ordinal not in range(128)
```

**Comment by alikins** [comment]
u'\u2018' is: ‘  ('LEFT SINGLE QUOTATION MARK' )  but not sure where that is coming from.

For the lxc system, what does the LANG (or other LC_\* ) env vars look like?

**Comment by bhcopeland** [comment]
LXC system, which is the same as my other servers:

```
LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```

**Comment by evillemez** [comment]
I think this is the same issue I am having.  I execute ansible from a VM via vagrant/virtualbox.  Environments are as follows:
- Host:
  - OS: Ubuntu 16.04.1 LTS
- Guest VM:
  - OS: Ubuntu 14.04.5 LTS
  - Python: 2.7.6
  - Ansible: 2.2.0.0
    - config file = /vagrant/ansible/ansible.cfg
    - configured module search path = Default w/o overrides
- Target Server:
  - OS: Ubuntu 14.04.4 LTS

The task in question hasn't changed in some time, but the VM was recently rebuilt, so there could be version differences for ansible/python between now and back when it worked for me previously.  Unfortunately I don't have a way to go back and see at what the versions were in the previously working state.  The host OS also changed from Mac OS X... though in theory that shouldn't be a factor.

If there's any other info I can provide that would be useful, please let me know!

Here's a traceback:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 119, in run
    res = self._execute()
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 490, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/python2.7/dist-packages/ansible/plugins/action/normal.py", line 33, in run
    results = merge_hash(results, self._execute_module(tmp=tmp, task_vars=task_vars))
  File "/usr/lib/python2.7/dist-packages/ansible/plugins/action/__init__.py", line 617, in _execute_module
    self._fixup_perms2(remote_files, remote_user)
  File "/usr/lib/python2.7/dist-packages/ansible/plugins/action/__init__.py", line 394, in _fixup_perms2
    ' see https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user'.format(res['rc'], res['stderr']))
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2018' in position 29: ordinal not in range(128)
```

**Comment by buunguyen** [comment]
Which version is this fix included in? I'm running 2.2.0.0_2 (installed via Brew) and still having the exact issue.

**Comment by enthurohini** [comment]
I freshly installed ansible version 2.2.0.0 on ubuntu 14.04 server, still having same problem. Any updates?

**Comment by abadger** [comment]
The fix for this is in 2.2.1 which is currently in a late release candidate stage. Take a look at https://releases.ansible.com/ansible/ for a 2.2.1 rc if you would like to test that this is fixed from a tarball or try running from a checkout of either the stable-2.2 or devel branches: http://docs.ansible.com/ansible/intro_installation.html#running-from-source
